src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/RouteStringSyntaxDetector.cs

```diff
@@ -164,6 +189,30 @@ private static bool IsAnyStringLiteral(int rawKind)
         }
 
         return false;
+
+        // Searching for the language comment on all parents is expensive (GetLeadingTrivia) if the token is deeply nested.
+        // An example of this is a string with hundreds of concatenations. Avoid this by checking the last inspected string node.
+        // The last inspected string will be a parent and allows us to exit quickly from a deeply nested tree.
+        static bool TryMatchPreviousStringNode(
+            SyntaxNode? node,
+            LastInspectedStringNode? lastInspectedStringNode,
+            out string? identifier,
+            out IEnumerable<string>? options)
+        {
+            if (lastInspectedStringNode is { } previousNode)
+            {
+                if (node != null && node == previousNode.Node.Parent)
```

Comment:
>node == previousNode.Node.Parent [](http://example.com/codeflow?start=36&length=32)

It looks like we're returning the identifier from the last string checked if the current node `node` is the parent of that last string node. If so, is that correct that the same identifier should be returned for all ancestors?

Reply:
That's how the current HasLanguageComment works. If any parent has a comment, then children inherit it.

However, there is a case where siblings can have different languages, e.g.

```cs
public static readonly string _test =
    // lang=regex
    "^(sdf)+" +
    // lang=Route
    "{sdf}";
```

And a child can override a parent, e.g. a comment on the field will be overriden by a comment on a string.

I'll add tests that these scenarios still work correctly.